src/test_application/DebugObject.h

```diff
@@ -48,23 +46,23 @@ class DebugObject : public AbstractDebugObject
 {
     friend Derived;
 public:
-    DebugObject(active::ActiveScene &scene, active::ActiveEnt ent) :
+    DebugObject(osp::active::ActiveScene &scene, osp::active::ActiveEnt ent) :
             m_scene(scene),
             m_ent(ent) {};
     virtual ~DebugObject() = default;
 
 private:
-    active::ActiveScene &m_scene;
-    active::ActiveEnt m_ent;
+    osp::active::ActiveScene &m_scene;
+    osp::active::ActiveEnt m_ent;
 };
 
-struct CompDebugObject
+struct ACompDebugObject
 {
-    CompDebugObject(std::unique_ptr<AbstractDebugObject> ptr) :
+    ACompDebugObject(std::unique_ptr<AbstractDebugObject> ptr) :
         m_obj(std::move(ptr)) {}
-    CompDebugObject(CompDebugObject&& move) = default;
-    ~CompDebugObject() = default;
-    CompDebugObject& operator=(CompDebugObject&& move) = default;
```

Comment:
Ok, sounds good. Many other ECS component structs in other places have non-constexpr move constructors too; these can be dealt with as well.

Reply:
I also recommend annotating functions with noexcept where possible.

That helps with code gen because the compiler doesn't need to worry about exception handling for that function call.

Reply:
The constexpr move constructor doesn't compile. "defaulted definition of move constructor is not constexpr"

Reply:
Need to do

constexpr CompDebugObject() = default;

= default; doesn't imply constexpr. (though, i wish it did).